### PR TITLE
feat: LT-2035 Generate documents when finalising

### PR DESCRIPTION
**Pull request by BenAndrew-Kainos**
Screenshot 2020-03-09 at 17 33 47

Key changes:
- Create FinalAdvice document endpoints extending base Generated document endpoints to maintain decision_key & modify fetch templates & post document calls
- Add Generate decision documents page

### CODE REVIEW

File: cases/views/advice.py [hunk 1]
```diff
@@ -363,4 +370,31 @@ def post(self, request, *args, **kwargs):
...
-        return redirect(reverse_lazy("cases:case", kwargs={"pk": case["id"]}))
+        return redirect(reverse_lazy("cases:finalise_documents", kwargs={"pk": case["id"]}))
+
+
+class FinaliseGenerateDocuments(TemplateView):
+    @staticmethod
+    def get_page(request, pk, errors=None):
+        decisions, _ = get_final_decision_documents(request, str(pk))
+        decisions = decisions["documents"]
+        can_submit = all([decision.get("document") for decision in decisions.values()])
```
Comment (stephanieboyd) [inline cases/views/advice.py]: This is that the licence can be submitted?

Comment (BenAndrew-Kainos) [reply cases/views/advice.py]: That the page can be submitted (not necessarily creating a licence)

File: ui_automation_tests/features/finalise_case.feature [hunk 2]
```diff
@@ -0,0 +1,49 @@
+@all @internal @finalise
+Feature: I want to finalise a case
+  As a logged in gov user
+  I want to be prompted to generate the right documents for the appropriate decision
+  So that this can be communicated back to the exporter and they can download and view their licence or clearance
+
+  @LT_2035_generate_documents_and_licence
```
Comment (stephanieboyd) [inline ui_automation_tests/features/finalise_case.feature]: Don't these need regression tags to run?

Comment (BenAndrew-Kainos) [reply ui_automation_tests/features/finalise_case.feature]: Yep my bad

Comment (stephanieboyd) [reply ui_automation_tests/features/finalise_case.feature]: I added them.

**Pull request merged**